File: javascript/001_Two_Sum.js
/// attempt 1
var twoSum = function(nums, target) {
    for(var i=0;i<nums.length;i++){
        var temp = target-nums[i]
        for (var x= 0; x<nums.length; x++ ){
            if(temp == nums[x] && x != i){
                return[i,x]
               }
        }
    }
};


// attempt 2
var twoSum = function(nums, target) {
    let obj = {};
    for(var i=0;i< nums.length;i++){
        let attamptNumber = target - nums[i];
        if (attamptNumber in obj){
            return [obj[attamptNumber] ,i]
        }
        obj[nums[i]] = i;
    }
};

File: python/001_Two_Sum.py
class Solution(object):

    def twoSum(self, nums, target):
        # two point
        afterOrder = enumerate(nums)
        for index, value in afterOrder:
            secondNum = target - value
            if secondNum in nums[index+1:]:
                    return [index, nums.index(secondNum,index+1)]
        return None
    
    def twoSumTemp2(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)):
            remaining = target - nums[i]
            if remaining not in dict:
                dict[nums[i]] = i
            else:
                return [ dict[remaining], i]
        return None
        

if __name__ == '__main__':
    # begin
    s = Solution()
    print s.twoSum([3, 2, 4], 6)
    print s.twoSum([3, 3], 6)
